electrumx/lib/atomicals_blueprint_builder.py

```diff
@@ -400,29 +448,33 @@ def calculate_nft_atomicals_splat(cls, nft_atomicals, tx):
             output_colored_map[expected_output_index]['atomicals'][atomical_id] = atomical_summary_info
             expected_output_index_incrementing += 1
         return AtomicalNftOutputBlueprintAssignmentSummary(output_colored_map)
-    
+
     @classmethod
     def custom_color_nft_atomicals(cls, nft_atomicals, operations_found_at_inputs, tx):
-      # define op `z` to custom-color nft
-      output_colored_map = {}
-      for atomical_id, atomical_info in sorted(nft_atomicals.items()):
-        for out_idx, txout in enumerate(tx.outputs):
-          compact_atomical_id = location_id_bytes_to_compact(atomical_id)
-          compact_atomical_id_data = operations_found_at_inputs["payload"].get(compact_atomical_id, {})
-          # if payload try to color two or more outputs, it will try to color output 0.
-          if len(compact_atomical_id_data.keys()) > 1:
-            expected_output_index = 0
-          else:
-            # if out_idx not in payload keys, skip
-            if str(out_idx) not in compact_atomical_id_data.keys():
-              continue
-            # if payload value <= 0, this nft will burn
-            if compact_atomical_id_data.get(str(out_idx), 0) <= 0:
-              continue
-            expected_output_index = out_idx
-          output_colored_map[expected_output_index] = output_colored_map.get(expected_output_index) or {'atomicals': {}}
-          output_colored_map[expected_output_index]['atomicals'][atomical_id] = atomical_info
-      return AtomicalNftOutputBlueprintAssignmentSummary(output_colored_map)
+        nfts_burned = {}
+        output_colored_map = {}
+        for atomical_id, atomical_info in sorted(nft_atomicals.items()):
+            remaining_value = atomical_info.atomical_value
+            for out_idx, txout in enumerate(tx.outputs):
+                compact_atomical_id = location_id_bytes_to_compact(atomical_id)
+                compact_atomical_id_data = operations_found_at_inputs["payload"].get(compact_atomical_id, {})
+                expected_value = compact_atomical_id_data.get(str(out_idx), 0)
+                if expected_value <= 0 or remaining_value <= 0:
+                    continue
+                # if payload try to color two or more outputs, it will try to color output 0.
+                if len(compact_atomical_id_data.keys()) > 1:
+                    expected_output_index = 0
+                else:
+                    if str(out_idx) not in compact_atomical_id_data.keys():  # if out_idx not in payload keys, skip
+                        continue
+                    expected_output_index = out_idx
+                if not output_colored_map.get(expected_output_index):
+                    output_colored_map[expected_output_index] = {'atomicals': {}}
+                output_colored_map[expected_output_index]['atomicals'][atomical_id] = atomical_info
+                remaining_value -= expected_value
+            if remaining_value > 0:
```

Comment:
I think no need use `remaining_value`, Nft should be full colored. If expected_value <= 0, Nft will burn.